uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-keycloak/src/main/java/org/uberfire/ext/security/management/keycloak/KeyCloakUserManager.java

```diff
@@ -82,109 +80,114 @@ public void initialize(final UserSystemManager userSystemManager) throws Excepti
         }
         final int page = req.getPage() - 1;
         final int pageSize = req.getPageSize();
-        RealmResource realmResource = getRealmResource();
-        UsersResource usersResource = realmResource.users();
-        List<UserRepresentation> userRepresentations = usersResource.search(req.getSearchPattern(),
-                                                                            page * pageSize,
-                                                                            pageSize + 1);
-
-        int numberOfFoundUsers = usersResource.search(req.getSearchPattern(),
-                                                      1,
-                                                      Integer.MAX_VALUE).size();
-
         final List<User> users = new ArrayList<User>();
-        boolean hasNext = false;
-        if (userRepresentations != null && !userRepresentations.isEmpty()) {
-            int x = 0;
-            for (UserRepresentation userRepresentation : userRepresentations) {
-                if (x == req.getPageSize()) {
-                    hasNext = true;
-                } else {
-                    final User user = createUser(userRepresentation);
-                    users.add(user);
-                    x++;
+        final int[] numberOfFoundUsers = new int[]{0};
```

Comment:
Why do you use int[] in this case instead of simple int?

Reply:
Because it's used later in a [lamda](https://github.com/kiegroup/appformer/pull/113/files/be16e5d502f4b07e215a90c11ead30953119c9fb#diff-23c89f77de0248bc83092de211684382R91). Using the "array approach" allows us to pass a ```final``` reference to inside the lamda and be able to change the actual _value_ (being element zero).

Reply:
Thanks :-)

Reply:
yeah it's how java works with anonymous functions.... the variables in the stack outside that function scope must be immutable,  the reference cannot change. Thx guys!